### PR TITLE
Add media item roles inside media items in product data.

### PR DESCRIPTION
**Pull request by nikunjkotecha**

### CODE REVIEW

**Comment by nikunjkotecha** [comment]
cc @sdelbosc @vbouchet31

File: Helper/Acm.php [hunk 1]
```diff
@@ -257,6 +264,13 @@ private function processMediaGalleryExtension()
             }
 
             if (isset($filterEntry['file'])) {
+                $filterEntry['roles'] = [];
+                foreach ($attributes as $role => $file) {
+                    if ($filterEntry['file'] == $file) {
+                        $filterEntry['roles'][] = $role;
+                    }
```
Comment (malachy-mcconnell) [inline Helper/Acm.php]: The array key is 'roles' but later, below, this function uses the service output processor to build the output data from the class `\Magento\Catalog\Api\ProductAttributeMediaGalleryManagementInterface::class`. Therefore I think the array key you want to use is `types` as per line 110 of file `vendor/magento/module-catalog/Api/Data/ProductAttributeMediaGalleryEntryInterface.php`

I mean I think the serviceOutputProcessor will drop any array keys that are not defined in `vendor/magento/module-catalog/Api/Data/ProductAttributeMediaGalleryEntryInterface.php`

What do you think? Oh, are we looking at different Magento versions? Is your Magento using `role` as the array key?

Comment (nikunjkotecha) [reply Helper/Acm.php]: Types is different, it is already set image, video, external_video

Comment (malachy-mcconnell) [reply Helper/Acm.php]: Interesting. I'm only going on the Magento function comment:
```
    /**
     * Retrieve gallery entry image types (thumbnail, image, small_image etc)
     *
     * @return string[]
     */
    public function getTypes();
```
Does the field 'types' not already hold the information you seek? small_image, image, thumbnail etc?

Anyway, really the question I am raising is, that I think if you want a new field 'role' it needs to be in the API data model for the serviceProcessor to include it, yes?

Comment (nikunjkotecha) [reply Helper/Acm.php]: `Does the field 'types' not already hold the information you seek? small_image, image, thumbnail etc?` - no, it contains image or video

`Anyway, really the question I am raising is, that I think if you want a new field 'role' it needs to be in the API data model for the serviceProcessor to include it, yes?`
There are open tickets already about this, as of today Magento doesn't provide this by default

**Comment by malachy-mcconnell** [comment]
I would like to leave this PR open until it also includes the API data model for `'role'`. Do you agree?

I at least want to create a new ticket 'add `'role'` to API data model' but I think it is simplest to add it here, yes?

**Pull request merged**